pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -478,12 +481,15 @@ func (s *DiscoveryServer) edsUpdate(clusterID, serviceName string, namespace str
 			if !f && !internal {
 				// The entry has a service account that was not previously associated.
 				// Requires a CDS push and full sync.
-				adsLog.Infof("Endpoint updating service account %s %s", e.ServiceAccount, serviceName)
+				adsLog.Debugf("Full push, new service account: svc = %v, service account %v", serviceName, e.ServiceAccount)
 				requireFull = true
-				break
 			}
 		}
 	}
+	if !reflect.DeepEqual(serviceAccounts, ep.ServiceAccounts) {
```

Comment:
after reading more of this code, i realize this might not be entirely correct. the endpoints shards.

the `s.EndpointShardsByService[serviceName][namespace] ` is a data structure as 

```go
ep = &EndpointShards{
			Shards:          map[string][]*model.IstioEndpoint{},
			ServiceAccounts: map[string]bool{},
		}
```

where the service account part is union of SA from all service discovery.
the input parameter `istioEndpoints` is just a update from a particular service discovery.

So the ep.ServiceAccounts can be superset of the SA from `istioEndpoints` legitimately. Blindingly overriding `ep.ServiceAccounts` is not appropriate, and can cause more full pushes than needed.

Reply:
after more thinking, i don't think this complexity of doing it strictly is worth it.